discovery/kubernetes/endpointslice.go

```diff
@@ -110,6 +110,9 @@ func NewEndpointSlice(l *slog.Logger, eps cache.SharedIndexInformer, svc, pod, n
 				e.logger.Error("converting to EndpointSlice object failed", "err", err)
 				continue
 			}
+			if es.Namespace != svc.Namespace {
+				continue
+			}
 			if lv, exists := es.Labels[v1.LabelServiceName]; exists && lv == svc.Name {
```

Comment:
let's not check for every endpointslices but only when we have a name match as in most cases the service informer would be namespaced.

```suggestion
                        // Only consider the underlying EndpointSlices in the same namespace.
			if svcName, exists := es.Labels[v1.LabelServiceName]; exists && svcName == svc.Name && es.Namespace == svc.Namespace {
```

Reply:
Thanks.